R/pivot-long.R

```diff
@@ -236,7 +236,12 @@ pivot_longer_spec <- function(data,
     # TODO somehow `t(matrix(x))` is _faster_ than `matrix(x, byrow = TRUE)`
     # if this gets fixed in R this should use `byrow = TRUE` again
     n_vals <- nrow(data) * length(val_cols)
-    idx <- t(matrix(seq_len(n_vals), ncol = n_vals / nrow(data)))
+    if (nrow(data) == 0L) {
+      ncol <- 0
+    } else {
+      ncol <- length(val_cols)
+    }
+    idx <- t(matrix(seq_len(n_vals), ncol = ncol))
```

Comment:
Can you just do this?

```r
idx <- t(matrix(seq_len(n_vals), ncol = length(val_cols)))
```

Previously you used `ncol = n_vals / nrow(data)`, but if `ncol = length(val_cols)` returns the correct result even in the 0-row case (since `val_cols` will be an empty list) then I think that would work nicely.

Reply:
Yes, it does work. I was a bit confused at first because we get back a matrix with dimensions but `as.integer()` handles this fine.